### PR TITLE
switching to ESM-by-default & upgrading actions [increment minor]

### PR DESCRIPTION
**Pull request by dlockhart**
This accomplishes two things that both needed to happen together:

### 1) New Visual Diff dependencies

The main motivation for this change. By switching to the new `main` branch of the actions repo, you pick up the latest visual-diff, mocha@9 and puppeteer@12 -- this will likely case minor visual differences.

### 2) Make this project use ES Modules by default

Mocha 9 (required for the new visual-diff action) requires it, and web projects shouldn't be CommonJS anyway. I've converted the things using CommonJS things (like `require()` and `__dirname`) to their ESM equivalents.

### Other minor updates

While I was in here, I updated a few things like the latest dev-server and test-runner and linting configs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,5 @@
 # @brightspace-ui/module-scheduler
 
-[![NPM version](https://img.shields.io/npm/v/@brightspace-ui/module-scheduler.svg)](https://www.npmjs.org/package/@brightspace-ui/module-scheduler)
-
```
Comment (dlockhart) [inline README.md:4]: This package doesn't publish to NPM.

File: README.md [hunk 2]
```diff
@@ -88,42 +92,3 @@ To start a [@web/dev-server](https://modern-web.dev/docs/dev-server/overview/) t
 ```shell
 npm start
 ```
-
-## Versioning & Releasing
```
Comment (dlockhart) [inline README.md:92]: This repo is using incremental-release not semantic, so I removed this section entirely.

File: package.json [hunk 3]
```diff
@@ -1,47 +1,42 @@
 {
   "name": "d2l-module-scheduler",
   "description": "Module Scheduler",
+  "type": "module",
   "repository": "https://github.com/BrightspaceUI/module-scheduler.git",
   "scripts": {
     "lint": "npm run lint:eslint && npm run lint:lit && npm run lint:style",
     "lint:eslint": "eslint . --ext .js,.html",
-    "lint:lit": "lit-analyzer copy-assignment-to-other-courses.js --strict",
```
Comment (dlockhart) [inline package.json:8]: Previously the Lit linting wasn't working as this must have been copy-pasted from another project.

File: src/components/module-scheduler-ignore-list.js [hunk 4]
```diff
@@ -95,7 +95,7 @@ class ModuleSchedulerIgnoreList extends BaseMixin(LocalizeMixin(LitElement)) {
 				attribute: false
 			},
 			_selectedItems: {
-				type: Set,
+				type: Object,
```
Comment (dlockhart) [inline src/components/module-scheduler-ignore-list.js:98]: Must be one of: `String`, `Number`, `Boolean`, `Array` or `Object`.

File: src/components/module-scheduler-manager.js [hunk 5]
```diff
@@ -330,7 +330,7 @@ class ModuleSchedulerManager extends BaseMixin(LocalizeMixin(LitElement)) {
 						${ this.isQuerying ? '' : this.allSchedules.map(schedule => this._renderSchedule(schedule)) }
 					</tbody>
 				</table>
-
+			</d2l-table-wrapper>
```
Comment (dlockhart) [inline src/components/module-scheduler-manager.js:333]: This closing tag that was missing ended up causing all the aXe tests to be skipped entirely.

File: src/components/module-scheduler-schedule-dialog.js [hunk 6]
```diff
@@ -266,7 +266,7 @@ class ScheduleDialog extends LocalizeMixin(LitElement) {
 				id="semesterId"
 				required
 				@change=${this._handleSemesterIdChange}
-				aria-invalid="${this._missingSemester}">
+				aria-invalid="${this._missingSemester ? 'true' : 'false'}">
```
Comment (dlockhart) [inline src/components/module-scheduler-schedule-dialog.js:269]: `aria-invalid` needs to be a string of `"true"` or `"false"`.

**Review by dbatiste — APPROVED** [review]
Wow... lots of fixes here!

**Pull request merged**